chain/messagepool/messagepool.go

```diff
@@ -31,17 +31,27 @@ import (
 
 var log = logging.Logger("messagepool")
 
-var (
-	ErrMessageTooBig = errors.New("message too big")
+const bigNonceThreshold uint64 = 100
 
-	ErrMessageValueTooHigh = errors.New("cannot send more filecoin than will ever exist")
+	type MpoolError struct {
+	msg   string
+	shouldBroadcast bool
+}
```

Comment:
I would use a sentinel error in combination with: `xerrors.Errorf("msg: %w")` and `xerrors.Is`.

It would work by defining:
`var ErrStillPush = erorrs.New("pushing despite error")`
using: `return xerrors.Errorf("not enough funds: %w", ErrStillPush)`
and checking with `xerrors.Is(err, ErrStillPush)`.

Name are subject to change.

Reply:
Done, it's much cleaner now, thanks!